system/metrics/sinks/elastic/elasticSink.cpp

```diff
@@ -43,8 +42,60 @@ ElasticMetricSink::ElasticMetricSink(const char *name, const IPropertyTree *pSet
     PeriodicMetricSink(name, "elastic", pSettingsTree)
 {
     ignoreZeroMetrics = pSettingsTree->getPropBool("@ignoreZeroMetrics", true);
-    pSettingsTree->getProp("@elasticHost", elasticHost);
-    pSettingsTree->getProp("@indexName", indexName);
+
+    StringBuffer hostName;
+    StringBuffer hostProtocol;
+    StringBuffer hostPort;
+
+    Owned<IPropertyTree> pHostConfigTree = pSettingsTree->getPropTree("host");
+    if (pHostConfigTree)
+    {
+        pHostConfigTree->getProp("@name", hostName);
+
+        if (!pHostConfigTree->getProp("@protocol", hostProtocol))
+        {
+            hostProtocol.append("https");
+        }
+
+        if (!pHostConfigTree->getProp("@port", hostPort))
+        {
+            hostPort.append("9200");
+        }
+    }
+
+    if (!hostName.isEmpty() && !hostPort.isEmpty() && !hostProtocol.isEmpty())
+    {
+        elasticHostUrl.append(hostProtocol).append("://").append(hostName).append(":").append(hostPort);
+    }
+    else
+    {
+        WARNLOG("ElasticMetricSink: Host configuration missing or invalid");
+    }
+
+    Owned<IPropertyTree> pIndexConfigTree = pSettingsTree->getPropTree("index");
+    if (pIndexConfigTree)
+    {
+        pSettingsTree->getProp("@name", indexName);
+    }
+
+    if (indexName.isEmpty())
+    {
+        WARNLOG("ElasticMetricSink: Index configuration missing or invalid");
+    }
+
+
+    // Both a host url and an index name are required
+    configurationValid = !elasticHostUrl.isEmpty() && !indexName.isEmpty();
+
+    // Initialize standard suffixes
+    countMetricSuffix.append("count");
+    gaugeMetricSuffix.append("gauge");
+    histogramMetricSuffix.append("histogram");
+
+    // See if any are overridden
+    pSettingsTree->getProp("@countMetricSuffix", countMetricSuffix);
```

Comment:
This code does not work.  getProp() appends to the existing value, it does not replace it.  I am not sure why this code uses a different pattern from the code above that gets the hostname.